libpdb/request_manager.go

```diff
@@ -7,6 +7,17 @@ import (
 	"time"
 )
 
+// The interface for the class that will determine what the next read should
+// be at each time stamp.
+type RequestGenerator interface {
```

Comment:
Same comment as above. How many classes need to implement this interface? Why can't request generation just happen in the request manager?